python_modules/libraries/dagster-fivetran/dagster_fivetran/v2/resources.py

```diff
@@ -0,0 +1,155 @@
+    destinations_by_id: Mapping[str, FivetranContentData]
+
+    @classmethod
+    def from_content_data(
+        cls, content_data: Sequence[FivetranContentData]
+    ) -> "FivetranWorkspaceData":
+        raise NotImplementedError()
+
+
+@experimental
+class FivetranClient:
+    """This class exposes methods on top of the Fivetran REST API."""
+
+    def __init__(
+        self,
+        api_key: str,
+        api_secret: str,
+        request_max_retries: int,
+        request_retry_delay: float,
+    ):
+        self.api_key = api_key
+        self.api_secret = api_secret
+        self.request_max_retries = request_max_retries
+        self.request_retry_delay = request_retry_delay
+
+    @property
+    def _auth(self) -> HTTPBasicAuth:
+        raise NotImplementedError()
+
+    @property
+    @cached_method
+    def _log(self) -> logging.Logger:
+        return get_dagster_logger()
+
+    @property
+    def api_base_url(self) -> str:
+        raise NotImplementedError()
+
+    @property
+    def api_connector_url(self) -> str:
+        raise NotImplementedError()
+
+    def make_connector_request(
+        self, method: str, endpoint: str, data: Optional[str] = None
+    ) -> Mapping[str, Any]:
+        raise NotImplementedError()
+
+    def make_request(
+        self, method: str, endpoint: str, data: Optional[str] = None
+    ) -> Mapping[str, Any]:
+        raise NotImplementedError()
+
+    def get_connector_details(self, connector_id: str) -> Mapping[str, Any]:
+        """Fetches details about a given connector from the Fivetran API."""
+        raise NotImplementedError()
+
+    def get_connectors_for_group(self, group_id: str) -> Mapping[str, Any]:
+        """Fetches all connectors for a given group from the Fivetran API."""
+        raise NotImplementedError()
+
+    def get_destination_details(self, destination_id: str) -> Mapping[str, Any]:
+        """Fetches details about a given destination from the Fivetran API."""
+        raise NotImplementedError()
+
+    def get_groups(self) -> Mapping[str, Any]:
+        """Fetches all groups from the Fivetran API."""
+        raise NotImplementedError()
+
+
+class FivetranWorkspace(ConfigurableResource):
+    """This class represents a Fivetran workspace and provides utilities
+    to interact with Fivetran APIs.
+    """
+
+    api_key: str = Field(description="The Fivetran API key to use for this resource.")
+    api_secret: str = Field(description="The Fivetran API secret to use for this resource.")
+    request_max_retries: int = Field(
+        default=3,
+        description=(
+            "The maximum number of times requests to the Fivetran API should be retried "
+            "before failing."
+        ),
+    )
+    request_retry_delay: float = Field(
+        default=0.25,
+        description="Time (in seconds) to wait between each request retry.",
+    )
+
+    _client: FivetranClient = PrivateAttr(default=None)
+
+    def get_client(self) -> FivetranClient:
```

Comment:
Curious what the motivation for separating out the client is here - just to separate out the workspace loading and the actual API interactions?

Reply:
Exactly - this is the current pattern for Tableau and Looker.

Reply:
Makes sense to me 👍